app/src/main/java/de/tum/in/tumcampusapp/activities/SurveyActivity.java

```diff
@@ -440,21 +452,76 @@ protected void onPostExecute(Void v) {
                         }.execute();
 
                     } else {
-                        // Without newly created questions
-                        setUpResponseTab();
+                        setUpResponseTab(); // setup responses tab Without possible fresh answeres.
                     }
                 }
             }
         });
     }
-    private void restartActivity() {
-        Intent i = getIntent();
-        i.addFlags(Intent.FLAG_ACTIVITY_NO_ANIMATION);
-        finish();
-        startActivity(i);
+
+
+    /**
+     * Help function for plausibility check of the question form
+     *
+     * @return true if everything was entered correctly else snackbar for requesting to complete questions.
+     */
+    private boolean checkSurveyData() {
+        boolean done = true;
```

Comment:
Instead of returning a boolean here, why not return the form data in an array or appropriate model? Using class variables increases the complexity and reduces readability of the code.